### ISSUE TITLE
Regression on version 0.16.9: expectation not found

### ISSUE THREAD

**Issue by Willem-J-an** [great-expectations/great_expectations#7766]
**Describe the bug**
Can not add expectations in version 0.16.9

**To Reproduce**
1. Install version 0.16.9
2. Create expectation suite
3. Add an expectation

``` python
suite = ExpectationSuite("Name")
expectations = [
    ExpectationConfiguration(
      expectation_type="expect_table_columns_to_match_ordered_list",
      kwargs={ "column_list": ["col1", "col2"] },
   )
]
for expectation in expectations:
  suite.add_expectation(expectation)
return suite
```

**Expected behavior**
Version 0.16.8 adds the expectation to the suite.
Version 0.16.9 throws an error, because it tries to validate the expectation name exists as key in an empty dictionary.
``` 
Traceback (most recent call last):
  File "/opt/conda/envs/dbconnect/lib/python3.9/site-packages/great_expectations/core/expectation_suite.py", line 753, in _validate_expectation_configuration_before_adding
    class_ = get_expectation_impl(expectation_configuration.expectation_type)
  File "/opt/conda/envs/dbconnect/lib/python3.9/site-packages/great_expectations/expectations/registry.py", line 406, in get_expectation_impl
    raise gx_exceptions.ExpectationNotFoundError(f"{expectation_name} not found")
great_expectations.exceptions.exceptions.ExpectationNotFoundError: expect_table_columns_to_match_ordered_list not found

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
    suite.add_expectation(expectation)
  File "/opt/conda/envs/dbconnect/lib/python3.9/site-packages/great_expectations/core/expectation_suite.py", line 739, in add_expectation
    self._validate_expectation_configuration_before_adding(
  File "/opt/conda/envs/dbconnect/lib/python3.9/site-packages/great_expectations/core/expectation_suite.py", line 760, in _validate_expectation_configuration_before_adding
    raise gx_exceptions.InvalidExpectationConfigurationError(
great_expectations.exceptions.exceptions.InvalidExpectationConfigurationError: Could not add expectation; provided configuration is not valid: expect_table_columns_to_match_ordered_list not found
```

**Environment (please complete the following information):**
 - Operating System: Linux
 - Great Expectations Version: 0.16.9

**Additional context**
As said this did not happen before 0.16.9.

**Comment by rafaelmatias2525** [comment]
The same is happening here. I just got an ExpectationNotFoundError.

**Comment by LukeHendrick** [comment]
Just adding a +1 - had to pin my Glue jobs to 0.16.8 to workaround

**Comment by austiezr** [comment]
Hey folks! We're investigating this now; if you're able, please share the full scripts generating this behavior (including imports, etc.), as well as your environment & OS details. That will help to accelerate a resolution here. Thank you! 🚀 

@rafaelmatias2525 @LukeHendrick @Willem-J-an

**Comment by LukeHendrick** [comment]
@austiezr - here's what I am using to reproduce (basically the same as @Willem-J-an's example):

```python
from great_expectations.core import ExpectationSuite, ExpectationConfiguration

suite = ExpectationSuite(expectation_suite_name="test_suite")
expectation = ExpectationConfiguration(expectation_type="expect_table_column_count_to_be_between", kwargs= {"min_value": 10, "max_value": 100}, meta={})

suite.add_expectation(expectation)
InvalidExpectationConfigurationError: Could not add expectation; provided configuration is not valid: expect_table_column_count_to_be_between not found
```

I have had this issue in AWS Glue 3.0/4.0 (Spark 3.1.1 and 3.3 respectively), as well as on my local machine from a new poetry module (Intel MacBook Pro,  macOS 13.3.1, Python 3.11)

```bash
poetry new expectations_repro
cd expectations_repro
poetry add great_expectations
poetry shell
python
```
And then in the REPL:
```python
import great_expectations
from great_expectations.core import ExpectationSuite, ExpectationConfiguration

suite = ExpectationSuite(expectation_suite_name="test_suite")
expectation = ExpectationConfiguration(expectation_type="expect_table_column_count_to_be_between", kwargs= {"min_value": 10, "max_value": 100}, meta={})

suite.add_expectation(expectation)
```

**Comment by austiezr** [comment]
Closing as resolved by #7771; this is [now live](https://greatexpectationstalk.slack.com/archives/C050KHMU3M3/p1682720908171239) in `0.16.10`. Thank you all for your help and your patience! 🚀
